lib/msc_extensions/msc_4186_simplified_sliding_sync/models/maybe.dart

```diff
@@ -0,0 +1,21 @@
+/// Used to depict the difference between null and undefined in JavaScript
+abstract class Maybe<T> {}
```

Comment:
Interesting concept. What do you think about we make this a `sealed` class? Then we could even use `switch` like:

```dart
extension on Map<String,Object?> {
  Maybe<T> getMaybe<T>(String key) {
    if (!containsKey(key)) return Undefined<T>();
    return Some<T>(this[key] as T?);
  }
}


final maybe = {'value': 5}.getMaybe<int>('value');
switch (maybe) {
  case Some: print('This is something'); break;
  case Undefined: print('This is undefined'); break;
}
```